docs/developers/deprecating.md

```diff
@@ -0,0 +1,45 @@
+#
+
+## What is deprecation?
+
+Whenever a feature of iRODS needs to be removed for whatever reason, we follow a process which first deprecates the feature and then removes it. Here are some possible reasons for deprecating a feature...
+
+ - It has some issue which can only be fixed by replacement.
+ - It has been or is being replaced by a better alternative.
+ - Its purpose is unknown or no longer supported.
+
+In general, users and administrators should expect the following process when a server feature is being deprecated:
+
+ 1. A feature can be declared as deprecated in any minor version.
+ 2. The feature will continue to be supported for all major versions in which it is deprecated (and previous major versions in which the feature exists).
+ 3. The feature can be removed in the next major version, but will not necessarily be removed. It may remain in a deprecated state in the next major version.
+ 4. If a feature is shown to be worth keeping after it has been deprecated and before it has been removed, it can be de-deprecated in any minor version.
```

Comment:
I snooped around a little bit and it seems that de-/un-deprecating something is seen as something of an anti-pattern, so maybe we should just... not mention this?

Reply:
I don't think we need to state that. As the consortium, if we determine we made an error in deprecating something, we can reverse it or leave it in a deprecated state. Deprecation doesn't necessarily mean we must remove the tool. It can also mean there are better alternatives for doing the same thing.

Of course, the goal is to never end up in that position.

Reply:
agreed.  let's not mention it here and then we continue to try to do good.

Reply:
Done